libcloud/compute/drivers/dimensiondata.py

```diff
@@ -1707,11 +1720,31 @@ def ex_remove_storage_from_node(self, node, disk_id):
 
         :rtype: ``bool``
         """
-        result = self.connection.request_with_orgId_api_1(
-            'server/%s/disk/%s?delete' %
-            (node.id, disk_id)).object
-        response_code = findtext(result, 'result', GENERAL_NS)
-        return response_code in ['IN_PROGRESS', 'SUCCESS']
+        disk = [disk for disk in node.extra['disks']
+                if disk.scsi_id == disk_id][0]
+        return self.ex_remove_storage(disk.id)
+
+    def ex_remove_storage(self, disk_id):
+        """
+        Remove storage from a node
+
+        :param  node: The server to add storage to
+        :type   node: :class:`Node`
+
+        :param  disk_id: The ID of the disk to remove
+        :type   disk_id: ``str``
+
+        :rtype: ``bool``
+        """
+        update_node = ET.Element('removeDisk',
```

Comment:
probably copy pasta from elsewhere, but update node is an odd name for this xml.